### PR TITLE
Add brok when start and and acknowledge and downtime

### PR DESCRIPTION
**Pull request by ddurieux**

### CODE REVIEW

File: alignak/acknowledge.py [hunk 1]
```diff
@@ -96,3 +97,35 @@ def serialize(self):
                 'end_time': self.end_time, 'author': self.author, 'comment': self.comment,
                 'persistent': self.persistent
                 }
+
+    def get_and_start_brok(self, host_name, service_name=''):
```
Comment (mohierf) [inline alignak/acknowledge.py]: Why not simply `get_raise_brok`?

File: alignak/acknowledge.py [hunk 2]
```diff
@@ -96,3 +97,35 @@ def serialize(self):
...
+        """
+        data = {'uuid': self.uuid}
+        data['host'] = host_name
+        if service_name != '':
+            data['service'] = service_name
+
+        brok = Brok({'type': 'acknowledge_raise', 'data': data})
+        return brok
+
+    def get_and_expire_brok(self, host_name, service_name=''):
```
Comment (mohierf) [inline alignak/acknowledge.py]: And why not simply `get_expire_brok` ?

File: alignak/acknowledge.py [hunk 3]
```diff
@@ -96,3 +97,35 @@ def serialize(self):
...
+        :param comment_type: 1 = host, 2 = service
+        :param host_name:
+        :param service_name:
+        :return: brok with wanted data
+        :rtype: alignak.brok.Brok
+        """
+        data = {'uuid': self.uuid}
+        data['host'] = host_name
+        if service_name != '':
+            data['service'] = service_name
```
Comment (mohierf) [inline alignak/acknowledge.py]: `data` should also contain the acknowledge specific parameters: sticky, persistent, author, comment, ...

File: alignak/downtime.py [hunk 4]
```diff
@@ -354,15 +363,34 @@ def fill_data_brok_from(self, data, brok_type):
                 if brok_type in entry['fill_brok']:
                     data[prop] = getattr(self, prop)
 
-    def get_initial_status_brok(self):
-        """Get a initial status brok
+    def get_and_start_brok(self, host_name, service_name=''):
```
Comment (mohierf) [inline alignak/downtime.py]: Same as for acknowledge: `get_downtime_raise`...

File: alignak/downtime.py [hunk 5]
```diff
@@ -354,15 +363,34 @@ def fill_data_brok_from(self, data, brok_type):
...
 
+        :param comment_type: 1 = host, 2 = service
+        :param host_name:
+        :param service_name:
         :return: brok with wanted data
         :rtype: alignak.brok.Brok
-        TODO: Duplicate from Notification.fill_data_brok_from
         """
         data = {'uuid': self.uuid}
+        data['host'] = host_name
```
Comment (mohierf) [inline alignak/downtime.py]: Same as for acknowledge: missing data ...

File: alignak/downtime.py [hunk 6]
```diff
@@ -354,15 +363,34 @@ def fill_data_brok_from(self, data, brok_type):
...
+        if service_name != '':
+            data['service'] = service_name
 
-        self.fill_data_brok_from(data, 'full_status')
         brok = Brok({'type': 'downtime_raise', 'data': data})
         return brok
+
+    def get_and_expire_brok(self, host_name, service_name=''):
```
Comment (mohierf) [inline alignak/downtime.py]: Same as for acknowledge: `get_downtime_expire`...

**Comment by mohierf** [comment]
Rebase / squash before merge please. Or, perharps we should merge only when the broker module will be finished ... if you ever need a small modification here ?

**Comment by ddurieux** [comment]
I rebase / squash to merge now

**Pull request merged**